src/dataframe/dataframe.jl

```diff
@@ -1278,12 +1334,19 @@ and columns are matched by order of appearance. In this case `row` must contain
 the same number of elements as the number of columns in `df`.
 
 If `row` is a `DataFrameRow`, `NamedTuple` or `AbstractDict` then
-values in `row` are matched to columns in `df` based on names (order is ignored).
-`row` may contain more columns than `df` if `columns=:intersect`
-(this is currently the default, but will change in the future), but all column names
-that are present in `df` must be present in `row`.
-Otherwise if `columns=:equal` then `row` must contain exactly the same columns as `df`
-(but possibly in a different order).
+values in `row` are matched to columns in `df` based on names. The exact behavior
+depends on `cols` argument value in the following way:
+* If `cols=:equal` (this is currently the default, but will change to `:identical` in the future)
+  then `row` must contain exactly the same columns as `df` (but possibly in a different order).
+* If `cols=:identical` then `row` must contain the same columns in the same order
+  (this option is the same as `:equal` if `row` is an `AbstractDict`).
+* If `cols=:intersect` then `row` may contain more columns than `df`,
+  but all column names that are present in `df` must be present in `row`.
+* If `cols=:subset` then `push!` behaves like for `:intersect` but if some column
+  is missing in `row` then a `missing` value is pushed to `df`.
+* Finally, if `cols=:union` then `push!` behaves like for `:subset` with two additional changes:
```

Comment:
"behaves like for `:subset`" is a bit hard to follow given that the latter is defined based on `:intersect`. :-) Maybe start the description from scratch?

Reply:
I leave this for a separate PR.